src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -685,6 +685,110 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
                 self.fail("Callback was not called")
 
 
+class _LoggingServerInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, tag, record):
+        self.tag = tag
+        self.record = record
+
+    async def intercept_service(self, continuation, handler_call_details):
+        self.record.append(self.tag + ':intercept_service')
+        return await continuation(handler_call_details)
+
+
+class _GenericServerInterceptor(aio.ServerInterceptor):
+
+    def __init__(self, fn):
+        self._fn = fn
+
+    async def intercept_service(self, continuation, handler_call_details):
+        return await self._fn(continuation, handler_call_details)
+
+
+def _filter_server_interceptor(condition, interceptor):
```

Comment:
To improve the maintainability of test cases, we might want to add type annotations, especially for callbacks.